### ISSUE TITLE
Need support for CCW X'F7' = DSO (Define Subsystem Operation)

### ISSUE THREAD

**Issue by Fish-Git** [SDL-Hercules-390/hyperion#547]
[In the main Hercules-390 support group, Aaron Finerman wrote on way back on Thu 12/23/2021 7:49 PM (in the "z/OS 2.4 JES2 must be cold started every time" thread):](https://hercules-390.groups.io/g/group/message/4280)

> ```
> 22:15:53 HHC01315I 0:6002 CHAN: ccw F760000C 7D3F1F58=>18000000 00004100 00000000          ............
> 22:15:53 HHC01312I 0:6002 CHAN: stat 0E00, count 0000
> 22:15:53 HHC01313I 0:6002 CHAN: sense 80000000 00FFFF01 00000000 00000000 00000000 00000000 00000080 00000000
> 22:15:53 HHC01314I 0:6002 CHAN: sense CMDREJ
> ```
> 
> The error during initialization is a 'DSO (Define Subsystem Operation x'F7') which Hercules does not support.
> 
> The error at the time of the failure was a 'Unit Check/No Record Found'.

&nbsp;
Then, on Sat 12/25/2021 12:50 PM, he wrote in a offline email to someone else which he forwarded a copy to me as an FYI courtesy:

> This is based on the latest DS8xxx manuals that IBM only distribues to other hardware vendors for a fee. Each page has the vendor's name watermarked, so if it leaks they know who did it.
> 
> DSO is very similar to PSF (Perform Subsystem Function x'27'). New orders and suborders are only added to DSO and PSF is a thing of the past.
> 
> Looks like z/OS is issuing order 'Prepare for Read Subsystem Data (x'18'), suborder 'Read Feature Codes x'41'.
> 
> DSO also speeds up IPL process when thousands of devices are involved, as it penetrates a reserve.
> 
> If I was in Hercules development, I would treat DSO the same as a PSF, because most older orders are the same format.

&nbsp;
This GitHub Issue is being created to formally document the above so we don't lose track of it, and in the hope that _some day_, one of us will have the time to try and implement the above suggested handling of the missing CCW opcode X'F7' support.

**Comment by salva-rczero** [comment]
I can confirm z/OS 2.5 ICKDSF Init vol utility, al use a F7 cmd associated with a 3E:
```
CCW CHAIN         FORMAT 1          IO                DEV..... 01C12    ASCB.... 00FB5E80 CPU..... 0
04ACAD80 F760000C 37DEBF58 18000000 00004100 00000000                                              |
04ACAD88 3E200100 37DEBDF8 00070009 000D0011 DDFFFDE5 E679BFD5 FFE00000 00000000 FFDCFFBE F7A00000 |
                           F7B7FFD2 80000000 AED00000 00000000 0059000D 00070000 F6DFFF80 00000000 |
```

**Comment by Fish-Git** [comment]
Closed by commit a047205d82e6cfbaff9df0cb5c4e958407c72948.
